File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/UnpackProgressUpdateListener.java
package com.mapbox.services.android.navigation.v5.navigation;

/**
 * Updates any UI elements on the status of the TAR unpacking.
 */
class UnpackProgressUpdateListener implements UnpackUpdateTask.ProgressUpdateListener {

  private final RouteTileDownloadListener listener;

  UnpackProgressUpdateListener(RouteTileDownloadListener listener) {
    this.listener = listener;
  }

  @Override
  public void onProgressUpdate(Long progress) {
    if (listener != null) {
      listener.onProgressUpdate(progress.intValue());
    }
  }

  @Override
  public void onCompletion() {
    listener.onCompletion();
  }
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/UnpackProgressUpdateListener.kt
package com.mapbox.services.android.navigation.v5.navigation

/**
 * Updates any UI elements on the status of the TAR unpacking.
 */
internal class UnpackProgressUpdateListener(private val listener: RouteTileDownloadListener) :
    UnpackUpdateTask.ProgressUpdateListener {

    override fun onProgressUpdate(progress: Long) {
        listener.onProgressUpdate(progress.toInt())
    }

    override fun onCompletion() {
        listener.onCompletion()
    }
}
